sync/handler_hello.go

```diff
@@ -38,6 +39,14 @@ func (handler *helloHandler) ParseMessage(m message.Message, initiator peer.ID)
 		return handler.acknowledge(response, initiator)
 	}
 
+	if time.Since(msg.MyTime()) > 10*time.Second {
+		response := message.NewHelloAckMessage(message.ResponseCodeRejected,
+			fmt.Sprintf("hello message difference time is more than 10 seconds,"+
```

Comment:
Messages might be slow to reach the node, so let's just say the times on the two nodes are different by more than 10 seconds: `time discrepancy exceeds 10 seconds`.